circuit_knitting/cutting/cutqc/dynamic_definition.py

```diff
@@ -0,0 +1,499 @@
+# This code is a Qiskit project.
+
+# (C) Copyright IBM 2022.
+
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# This code was ported from CutQC at https://github.com/weiT1993/CutQC
+# (C) Copyright 2022 Wei Tang (weit@princeton.edu)
```

Comment:
```suggestion
# The code in this file was ported from CutQC at https://github.com/weiT1993/CutQC
# (C) Copyright 2022 Wei Tang (weit@princeton.edu), available under the MIT License.
```

Reply:
If I take the [license](https://github.com/weiT1993/CutQC/blob/main/LICENSE) literally, the full text of the license should be added somewhere to this repository, as well.  One way to do that would be to put the license text in a comment block following this line.

Reply:
Thanks for the correction, I added the URL to the code instead of including the full text of the MIT License. I found several codes that only mention the URL in the same way, so I think this will work.

Reply:
Actually, for our purposes, I think it would be better to include the full text of the license in the file.  There is an example of this being properly documented in a rustworkx file: https://github.com/Qiskit/rustworkx/blob/main/rustworkx/visualization/matplotlib.py